sdk/appcenter/src/main/java/com/microsoft/appcenter/http/HttpClientRetryer.java

```diff
@@ -92,7 +94,16 @@ public synchronized void cancel() {
         public void onCallFailed(Exception e) {
             if (mRetryCount < RETRY_INTERVALS.length && HttpUtils.isRecoverableError(e)) {
                 long delay = RETRY_INTERVALS[mRetryCount++] / 2;
-                delay += mRandom.nextInt((int) delay);
+
+                if (e instanceof HttpException) {
+                    HttpException httpException = (HttpException) e;
+                    if (!httpException.getHeaders().isEmpty() && httpException.getHeaders().containsKey(RETRY_AFTER_MS)) {
```

Comment:
if you exit this second if, then delay will not have the increment as a fallback.

Consider refactoring:
```
long delay = 0;
first if http {
     second if http {
      }
}
if (delay is still 0) { 
      long delay = RETRY_INTERVALS[mRetryCount++] / 2;
      delay += mRandom.nextInt((int) delay);
}
```

Reply:
I don't see the fix but comment was resolved.